scripts/aggregate_cells.py

```diff
@@ -98,18 +100,22 @@ def aggregate(
     aggregate_profile_out = [
         str(out_name) for out_name in snakemake.output["aggregate_profile"]
     ]
-    meta_data_dir = str(snakemake.input["metadata"])
-    barcode = str(snakemake.input["barcodes"])
-    config_path = str(snakemake.params["aggregate_config"])
-    inputs = zip(sqlfiles, cell_count_out, aggregate_profile_out)
+    meta_data_dir = itertools.repeat(str(snakemake.input["metadata"]))
+    barcode_map = itertools.repeat(str(snakemake.input["barcodes"]))
+    config_path = itertools.repeat(str(snakemake.params["aggregate_config"]))
+    inputs = list(
+        zip(sqlfiles, meta_data_dir, barcode_map, cell_count_out, aggregate_profile_out, config_path)
+    )
 
-    # running the aggregate algorithm
-    for sqlfile, cell_count_out, aggregate_profile_out in inputs:
-        aggregate(
-            sql_file=sqlfile,
-            metadata_dir=meta_data_dir,
-            barcode_path=barcode,
-            aggregate_file_out=aggregate_profile_out,
-            cell_count_out=cell_count_out,
-            config=config_path,
+    # init multi process
+    n_cores = int(snakemake.threads)
+    if n_cores > len(inputs):
+        print(
+            f"WARNING: number of specify cores exceeds number of inputs, defaulting to {len(inputs)}"
```

Comment:
```suggestion
            f"WARNING: number of specified cores ({n_cores}) exceeds number of inputs ({len(inputs)}), defaulting to {len(inputs)}"
```

Also, two other discussion items:

1. What is the difference between cores and threads? Should we keep the name consistent?
2. Maybe instead of a print statement, we can use the [`warnings` package](https://docs.python.org/3/library/warnings.html). This might be a better fit once we convert to logging in #9

Reply:
Realistically, it is cores. This is because if we used threads, we're still dealing with the Global Interpreter Lock (GIL); therefore, removing parallelization :(. 

The multi processing spawns multiple `aggregate.py` processes  per core. 

However, this gets confusing because there is no `cores` rule attribute in `snakemake` only `threads` 

personally, I would like it to be set to `cores` because it reflects the actual truth what is being used in the machine. However, since we are limited to the parameter names in` snakemake`, it would be best to use `threads` in order to keep  it consistent and readable for users.